src/main/java/frc/robot/subsystems/Maflipulator.java

```diff
@@ -0,0 +1,129 @@
+        setDefaultCommand(manualFlip());
+    }
+
+    public Command manualFlip() {
+        // The command is named "Manual Flip" and requires this subsystem.
+        return new Command("Manual Flip", this) {
+            @Override
+            protected void initialize() {
+                // Called just before this Command runs the first time
+            }
+
+            @Override
+            protected void execute() {
+                flipMotor.set(Robot.coPilot.getY(Hand.kRight));
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    public Command restrictRotate() {
+        // The command is named "Restrict Rotate" and requires this subsystem.
+        return new Command("Restrict Rotate", this) {
+
+            @Override
+            protected void execute() {
+                double flipSpeed = Robot.coPilot.getY(Hand.kRight);
+                if (currentPosition == MaflipulatorSide.kFront) {
+                    if (flipSpeed > 0 && manipulatorPot.get() >= 180) {
+                        flipSpeed = 0;
+                    }
+                    if (flipSpeed < 0 && manipulatorPot.get() <= 70) {
+                        flipSpeed = 0;
+                    }
+                }
+
+                if (currentPosition == MaflipulatorSide.kBack) {
+                    if (flipSpeed > 0 && manipulatorPot.get() <= 180) {
+                        flipSpeed = 0;
+                    }
+                    if (flipSpeed < 0 && manipulatorPot.get() >= 290) {
+                        flipSpeed = 0;
+                    }
+                }
+                flipMotor.set(flipSpeed);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+
+            @Override
+            protected void end() {
+                // Called once after isFinished returns true
+            }
+        };
+    }
+
+    public Command Flip() {
+        // The command is named "Flip" and requires this subsystem.
+        return new Command("Flip", this) {
+            @Override
+            protected void initialize() {
+                if (currentPosition == MaflipulatorSide.kFront) {
+                    flipMotor.set(1);
+                } else {
+                    flipMotor.set(-1);
+                }
+            }
+
+            @Override
+            protected void execute() {
+                // Called repeatedly when this Command is scheduled to run
+            }
+
+            @Override
+            protected boolean isFinished() {
+                // Make this return true when this Command no longer needs to run execute()
+                if (currentPosition == MaflipulatorSide.kFront && manipulatorPot.get() >= 270) {
+                    return true;
+                }
+                if (currentPosition == MaflipulatorSide.kBack && manipulatorPot.get() <= 90) {
```

Comment:
Could just be ` else if (manipulatorPot.get() <= 90)` since there's only two possible sides. (We could probably just use a bool here. But enums are fun!